File: Week2/readability.c
#include <cs50.h>
#include <ctype.h>
#include <math.h>
#include <stdio.h>
#include <string.h>

void reading_level(string text);

int main()
{
    string text = get_string("Text: ");
    reading_level(text);
}

void reading_level(string text)
{
    int words = 1, letters = 0, sentences = 0;
    int len = strlen(text);
    for (int i = 0; i < len; i++)
    {
        if (isspace(text[i]))
        {
            words += 1;
        }
        else if (ispunct(text[i]))
        {
            if (text[i] == '!' || text[i] == '.' || text[i] == '?')
            {
                sentences += 1;
            }
        }
        else if (isalpha(text[i]))
        {
            letters += 1;
        }
    }
    float ind = 0.0, L = (100.0 / words * letters), S = (100.0 / words * sentences);
    ind = 0.0588 * L - 0.296 * S - 15.8;
    int index = floor(ind + 0.5);
    if (index > 16.0)
    {
        printf("Grade 16+\n");
    }
    else if (index < 1.0)
    {
        printf("Before Grade 1\n");
    }
    else
    {
        printf("Grade %i\n", index);
    }
}

File: Week6/readability.py
import string


def reading_level(text):
    words = 1
    letters = 0
    sentences = 0
    length = len(text)
    for i in range(0, length, 1):
        if text[i].isspace():
            words += 1
        elif text[i] in string.punctuation:
            if (text[i] in ['.', '?', '!']):
                sentences += 1
        elif text[i].isalpha():
            letters += 1
    L = (100.0 / words * letters)
    S = (100.0 / words * sentences)
    level = 0.0588 * L - 0.296 * S - 15.8
    if (level > 16.0):
        print("Grade 16+")
    elif (level < 1.0):
        print("Before Grade 1")
    else:
        print(f"Grade {round(level)}")


text = input("Text: ")
reading_level(text)
